numpy/f2py/crackfortran.py

```diff
@@ -1371,80 +1371,85 @@ def analyzeline(m, case, line):
+                value = c
+                init_list = ''
+                state = 0
+        if state == 2:
+            value = value.strip()
+            if value.startswith(','):
+                value = value[1:].strip()
+            values.append([value, init_list])
+        variables = {}
         if 'vars' in groupcache[groupcounter]:
-            vars = groupcache[groupcounter]['vars']
-        last_name = None
-        for l in ll:
-            l = [x.strip() for x in l]
-            if l[0][0] == ',':
-                l[0] = l[0][1:]
-            if l[0][0] == '(':
+            variables = groupcache[groupcounter]['vars']
+        last_variable = None
+        for pair in values:
+            pair = [x.strip() for x in pair]
+            if pair[0][0] == ',':
+                pair[0] = pair[0][1:]
+            if pair[0][0] == '(':
                 outmess(
-                    'analyzeline: implied-DO list "%s" is not supported. Skipping.\n' % l[0])
+                    'analyzeline: implied-DO list "%s" is not supported. Skipping.\n' % pair[0])
                 continue
-            i = 0
-            j = 0
-            llen = len(l[1])
-            for v in rmbadname([x.strip() for x in markoutercomma(l[0]).split('@,@')]):
-                if v[0] == '(':
+            index = 0
+            end_index = 0
+            llen = len(pair[1])
+            for name in rmbadname([x.strip() for x in markoutercomma(pair[0]).split('@,@')]):
+                if name[0] == '(':
                     outmess(
-                        'analyzeline: implied-DO list "%s" is not supported. Skipping.\n' % v)
+                        'analyzeline: implied-DO list "%s" is not supported. Skipping.\n' % name)
                     # XXX: subsequent init expressions may get wrong values.
                     # Ignoring since data statements are irrelevant for
                     # wrapping.
                     continue
-                fc = 0
-                while (i < llen) and (fc or not l[1][i] == ','):
-                    if l[1][i] == "'":
-                        fc = not fc
-                    i = i + 1
-                i = i + 1
-                if v not in vars:
-                    vars[v] = {}
-                if '=' in vars[v] and not vars[v]['='] == l[1][j:i - 1]:
+                quote_open = 0
+                try:
+                    pair_type = groupcache[groupcounter]['vars'][pair[0]]['typespec']
+                except KeyError:
+                    pair_type = None
+                while index < llen and (pair_type == 'complex' or (quote_open or pair[1][index] != ',')):
+                    if pair[1][index] == "'":
+                        quote_open = not quote_open
+                    index += 1
+                index = index + 1
+                if name not in variables:
+                    variables[name] = {}
+                if '=' in variables[name] and not variables[name]['='] == pair[1][end_index:index - 1]:
```

Comment:
This is the new bit? Could you explain what is going on here?

Reply:
The earlier loop conditional was:

```python
                while (i < llen) and (fc or not l[1][i] == ','):
```

Which is now essentially changed to:

```python
                while (i < llen):
```

When the type of the variable `groupcache[groupcounter]['vars'][pair[0]]['typespec']` is complex. The reason for this is that complex data statements have 2 values assigned to a single variable, and the older logic split at the comma to handle statements like `data x, y / 1.5, 2.0 /`.

Reply:
I just realized this will not work for assigning multiple complex variables in one statement. Will update this to parse sets of two values instead.